lib/brain/redis.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+require 'socket'
+require 'logger'
+require 'json'
+require 'yaml'
+require 'docker'
+require 'redis'
+
+# Brain Class
+# Its main functions are to:
+#  1. Create Redis Container
+#     - Local Docker Install or DIND depending on Setup
+#     - Validates if a previous container is running and replaces it
+#     - Mounts Path to DIND Host or Localhost for Redis AOF file (Data Persistance)
+#  2. Create Redis Client Access
+#     - URL For Redis determined by IP Comparison (Checking for Compose Environment)
+#  3. Enables Access to Brain
+#     - Query brain for specific key (i.e. Plugin User)
+#     - Query brain for specific hash (i.e. Plugin)
+#     - Delete Key from Brain (i.e. Plugin User)
+#     - Save Key/Value to Brain (i.e. Plugin User Bob )
+class Brain
+  def initialize(settings)
+    @logger = Logger.new(STDOUT)
+    @logger.level = settings.logger_level
+  end
+
+  def brain
+    brain_check('slapi_brain')
+    @image = Docker::Image.create(fromImage: 'redis:3-alpine')
+    @container_hash = {
+      'name' => 'slapi_brain',
+      'Image' => 'redis:3-alpine',
+      'Cmd' => ['redis-server', '--appendonly', 'yes'],
+      'ExposedPorts' => { '6379/tcp' => {} },
+      'HostConfig' => {
+        'PortBindings' => {
+          '6379/tcp' => [{ 'HostPort' => '6379', 'HostIp' => '0.0.0.0' }]
+        },
+        'Binds' => ["#{Dir.pwd}/brain/:/data"]
+      }
+    }
+    @container_hash['Entrypoint'] = @image.info['Config']['Entrypoint']
+    @container_hash['WorkingDir'] = @image.info['Config']['WorkingDir']
+    @container_hash['Labels'] = @image.info['Config']['Labels']
```

Comment:
Is there any thing in the @image.info['Config'] that should not be merged?

If not then use a simple Hash#merge. https://ruby-doc.org/core-2.4.0/Hash.html#method-i-merge.

You would have to play with this, but it would look something like:

    @container_hash.merge(@image.info['Config'])